webpack/InsightsCloudSync/InsightsCloudSync.js

```diff
@@ -1,57 +1,61 @@
-import InsightsSettings from './Components/InsightsSettings';
+import PageLayout from 'foremanReact/routes/common/PageLayout/PageLayout';
+import InsightsHeader from './Components/InsightsHeader';
+import {
+  INSIGHTS_SYNC_PAGE_TITLE,
+  INSIGHTS_SEARCH_PROPS,
+} from './InsightsCloudSyncConstants';
 
-const InsightsCloudSync = ({ settingsUrl, syncInsights }) => {
-  document.title = INSIGHTS_SYNC_PAGE_TITLE;
+const InsightsCloudSync = ({
+  settingsUrl,
+  syncInsights,
+  updateUrlQuery,
+  query,
+  fetchInsights,
+  page,
+  perPage,
+}) => {
+  useEffect(() => {
+    fetchInsights({ page, perPage, query });
+  }, [fetchInsights, page, perPage, query]);
   return (
-    <IntlProvider locale={navigator.language}>
-      <div className="insights-cloud-sync">
-        <h1>{__('Red Hat Insights Sync')}</h1>
-        <div className="insights-cloud-sync-body">
-          <InsightsSettings />
-          <p>
-            {__(`Insights synchronization process is used to provide Insights
-             recommendations output for hosts managed here`)}
-          </p>
-          <p>
-            {__(`1. Obtain an Red Hat API token: `)}
-            <a
-              href="https://access.redhat.com/management/api"
-              target="_blank"
-              rel="noopener noreferrer"
-            >
-              access.redhat.com <Icon name="external-link" size="xs" />
-            </a>
-            <br />
-            {__("2. Copy the token to 'Red Hat Cloud token' setting: ")}
-            <a href={settingsUrl} target="_blank" rel="noopener noreferrer">
-              {__('Red Hat Cloud token ')}
-              <Icon name="external-link" size="xs" />
-            </a>
-            <br />
-            {__(
-              '3. Now you can synchronize recommendations manually using the "Sync now" button.'
-            )}
-          </p>
-          <div>
-            <Button bsStyle="primary" onClick={syncInsights}>
-              {__('Sync now')}
-            </Button>
-          </div>
-        </div>
-      </div>
-    </IntlProvider>
+    <PageLayout
+      searchable
+      searchProps={INSIGHTS_SEARCH_PROPS}
+      onSearch={searchQuery => updateUrlQuery({ searchQuery })}
+      header={INSIGHTS_SYNC_PAGE_TITLE}
+      toolbarButtons={
+        <Button bsStyle="primary" onClick={syncInsights}>
+          {__('Sync now')}
+        </Button>
+      }
+      searchQuery={query}
+      beforeToolbarComponent={<InsightsHeader settingsUrl={settingsUrl} />}
+    >
+      <React.Fragment>
+        <p>Insights Table will be here</p>
+      </React.Fragment>
+    </PageLayout>
   );
 };
 
 InsightsCloudSync.propTypes = {
   syncInsights: PropTypes.func.isRequired,
   settingsUrl: PropTypes.string.isRequired,
+  updateUrlQuery: PropTypes.func.isRequired,
+  fetchInsights: PropTypes.func.isRequired,
+  page: PropTypes.number,
+  perPage: PropTypes.number,
+  query: PropTypes.string,
+};
+
+InsightsCloudSync.defaultProps = {
+  page: 1,
+  perPage: 25,
```

Comment:
Does it mean it will send `25` even if nobody set this value? It's not consistent with the server side setting.

@mariaaga, is there a better way to sync pagination properties with the server?

Reply:
I think we could use foreman context as default for pagination,
those values are initially set when the foreman app starts,
another option is to create an initial API call on our page in order to get those values.

Reply:
I think we need to use foreman's context. It already handles pagination values for us.

Reply:
done, using its value if `per_page` from `url` is `undefined`

Reply:
had another thought about it, 
in this case, the table won't cause the scrollbar to appear when there are 5 rows per page,
maybe 5 should be our default?
above that, it will make the user to scroll down.